controllers/awsmachine_controller.go

```diff
@@ -639,6 +648,45 @@ func (r *AWSMachineReconciler) reconcileOperationalState(ec2svc services.EC2Inte
 		return err
 	}
 
+	// check if the remote kubeconfig works and annotate the cluster
+	_, ok := machineScope.InfraCluster.InfraCluster().GetAnnotations()[scope.KubeconfigReadyAnnotation]
+	if !ok && machineScope.IsControlPlane() {
+		// if a control plane node is operational check for a kubeconfig and a working control plane node
+		// and set the annotation so any reconciliation which requires workload api access can complete
+		remoteClient, err := machineScope.InfraCluster.RemoteClient()
+		if err != nil {
+			return err
+		}
+
+		var nodes corev1.NodeList
+		if err := remoteClient.List(ctx, &nodes, client.MatchingLabels(map[string]string{"node-role.kubernetes.io/control-plane": ""})); err != nil {
+			return err
+		}
+
+		oneReady := false
+		for i := range nodes.Items {
+			if util.IsNodeReady(&nodes.Items[i]) {
+				oneReady = true // if one control plane is ready return true
+			}
+		}
+
+		if oneReady {
+			awsCluster := &infrav1.AWSCluster{}
+			key := types.NamespacedName{Namespace: machineScope.InfraCluster.Namespace(), Name: machineScope.InfraCluster.Name()}
+			if err := r.Client.Get(ctx, key, awsCluster); err != nil {
+				return err
+			}
+			anno := awsCluster.GetAnnotations()
+			anno[scope.KubeconfigReadyAnnotation] = "true"
+			awsCluster.SetAnnotations(anno)
+			if err := r.Client.Update(ctx, awsCluster); err != nil {
```

Comment:
@richardcase Is this the right way to update the annotation on aws Cluster object? Don't we have to use

```
func (s *ClusterScope) PatchObject() error {
```

... or something similar? Although if I'm not mistaken the patcher can't update annotations just the spec and status or something like that... Maybe I'm mistaken.